### PR TITLE
Use keyboard shortcut only if ACL allows it

### LINKED ISSUE #7910: A user should be able to use keyboard shortcuts only if ACLs allow it
### Is your feature request related to a problem? Please describe.

As a recipient of a shared mailbox I want the app to only show me options allowed by ACL

### Describe the solution you'd like

Change the front end for actions on keyboard shortcuts:
- Compose new message
- Toggle start
- Toggle unread
- Delete
- Send

### Describe alternatives you've considered

_No response_

### Additional context

This is part of https://github.com/nextcloud/mail/issues/7057

### PR DESCRIPTION
**Pull request by GretaD**
fixes #7910

### CODE REVIEW

**Comment by ChristophWurst** [comment]
We use https://www.npmjs.com/package/vue-shortkey for keyboard shortcuts. The directive is attached to `AppContentList`. `MailboxThread::onShortcut` handles the shortcut and emits an event. In ``Mailbox::handleShortcut`` you find the shortcut handler method. Adjust that method to ACLs.

File: src/components/Mailbox.vue [hunk 1]
```diff
@@ -351,7 +369,9 @@ export default {
 				break
 			case 'del':
 				logger.debug('deleting', { env })
-				this.onDelete(env.databaseId)
+				if (e.srcKey === 'del' && this.hasDeleteAcl()) {
```
Comment (ChristophWurst) [inline src/components/Mailbox.vue:372]: this code is inside a switch case where e.shortKey matches `del`. I don't see a need for a second check. The first part of the if condition will always be `true`

File: src/components/Mailbox.vue [hunk 2]
```diff
@@ -351,7 +369,9 @@ export default {
...
+					this.onDelete(env.databaseId)
+				}
 				try {
 					await this.$store.dispatch('deleteThread', {
```
Comment (ChristophWurst) [inline src/components/Mailbox.vue:376]: this still executes, even with lacking ACL rights, doesn't it?

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: src/components/Mailbox.vue (hunk 2 above)
Comment (GretaD) [reply src/components/Mailbox.vue:376]: Here i noticed a bug, the message is gone from the list giving you the impression that is deleted but after reload message it is still there.

File: src/components/Mailbox.vue [hunk 3]
```diff
@@ -407,12 +429,14 @@ export default {
 				break
 			case 'unseen':
 				logger.debug('marking message as seen/unseen via shortkey', { env })
```
Comment (ChristophWurst) [inline src/components/Mailbox.vue:431]: nit: move into the if

nit2: use an early return like above

File: src/views/KeyboardShortcuts.vue [hunk 4]
```diff
@@ -29,6 +29,10 @@
 				<dt><kbd>U</kbd></dt>
 				<dd>{{ t('mail', 'Toggle unread') }}</dd>
 			</div>
+			<div>
+				<dt><kbd>A</kbd></dt>
+				<dd>{{ t('mail', 'Toggle archiving') }}</dd>
```
Comment (ChristophWurst) [inline src/views/KeyboardShortcuts.vue:34]:
```suggestion
				<dd>{{ t('mail', 'Archive') }}</dd>
```

File: src/components/Mailbox.vue [hunk 5]
```diff
@@ -350,7 +368,11 @@ export default {
 				})
 				break
 			case 'del':
-				logger.debug('deleting', { env })
+				if (!this.hasDeleteAcl()) {
+					logger.debug('deleting', { env })
```
Comment (ChristophWurst) [inline src/components/Mailbox.vue:372]: it logs when it doesn't delete, and doesn't log when it does delete

File: src/components/Mailbox.vue [hunk 6]
```diff
@@ -374,7 +396,10 @@ export default {
 
 				break
 			case 'arch':
-				logger.debug('archiving', { env })
+				if (this.hasArchiveAcl()) {
+					logger.debug('archiving', { env })
```
Comment (ChristophWurst) [inline src/components/Mailbox.vue:400]: same here

File: src/components/Mailbox.vue [hunk 7]
```diff
@@ -406,13 +430,18 @@ export default {
 
 				break
 			case 'unseen':
-				logger.debug('marking message as seen/unseen via shortkey', { env })
+				if (this.hasSeenAcl()) {
+					logger.debug('marking message as seen/unseen via shortkey', { env })
```
Comment (ChristophWurst) [inline src/components/Mailbox.vue:434]: same

**Review by ChristophWurst — COMMENTED** [review]
Move the log *after* the if

**Pull request merged**